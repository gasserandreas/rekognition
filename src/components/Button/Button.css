.button {
  padding: .6rem .8rem;
  text-transform: uppercase;
  font-weight: 300;
  border-width: 1px;
  border-radius: 3px;
  font-size: .8rem;
  transition: color .15s ease-in-out,
    background-color .15s ease-in-out,
    border-color .15s ease-in-out,
    box-shadow .15s ease-in-out;
}

.button:hover {
  cursor: pointer;
}

.button.default {
  background-color: #fff;
  border-color: #999;
  color: #133545;
}

.button.default:hover {
  background-color: #F6F6F8;
}

.button.blue {
  background-color: #133545;
  border-color: #133545;
  color: #fff;
}

.button.blue:hover {
  background-color: #264d5d;
  background-color: #264d5d;
}

.button.green {
  background-color: #51B17F;
  border-color: #51B17F;
  color: #fff;
}

.button.green:hover {
  background-color: #41A16F;
}