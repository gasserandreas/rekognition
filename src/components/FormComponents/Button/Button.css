.button {
  display: inline-block;
  background: none;
  border-style: solid;
  border-width: 1px;
  border-radius: 50px;
  padding: 7px 16px;
  font-size: 14px;
  font-size: 1.4rem;
  font-weight: 700;
  line-height: 1.2;
  text-align: center;
  text-decoration: none;
  /* text-transform: lowercase; */
}

.button:hover {
  cursor: pointer;
}

.button[disabled] {
  cursor: not-allowed;
}

.button.default,
.button.default[disabled]:hover,
.button.default.filled:hover {
  border-color: #00a862;
  color: #00a862;
  background-color: inherit;
}

.button.default.filled,
.button.default.filled[disabled]:hover,
.button.default:hover {
  background-color: #00a862;
  color: #fff;
}