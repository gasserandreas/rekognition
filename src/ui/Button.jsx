import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import * as Grommet from 'grommet';

import { Colors } from '../styles';

const BUTTON_TYPES = {
  DEFAULT: 'default',
  PRIMARY: 'primary',
  WARNING: 'warning',
  ERROR: 'error',
  LINK: 'link',
};

// calculate background colors
const getBackgroundColor = props => {
  switch (props.buttonStyle) {
    case BUTTON_TYPES.PRIMARY:
      return Colors.Blue.Default;
    case BUTTON_TYPES.WARNING:
      return Colors.Orange.Default;
    case BUTTON_TYPES.ERROR:
      return Colors.Red.Default;
    case BUTTON_TYPES.LINK:
      return 'inherit';
    default:
      return 'rgba(9, 30, 66, 0.04)'
  }
}

const getBackgroundHoverColor = props => {
  switch (props.buttonStyle) {
    case BUTTON_TYPES.PRIMARY:
      return Colors.Blue.Light;
    case BUTTON_TYPES.WARNING:
      return Colors.Orange.Light;
    case BUTTON_TYPES.ERROR:
      return Colors.Red.Light;
    case BUTTON_TYPES.LINK:
      return 'inherit';
    default:
      return 'rgba(9, 30, 66, 0.08);';
  }
}

// text color
const getColor = props => {
  switch (props.buttonStyle) {
    case BUTTON_TYPES.PRIMARY:
    case BUTTON_TYPES.ERROR:
      return Colors.ColorsPalette.White;
    case BUTTON_TYPES.LINK:
      return Colors.Blue.Default;
    default:
      return '#494949';
  }
}

const StyledButton = styled(Grommet.Button)`
  border-radius: 3px;
  font-weight: 400;
  height: 34px;
  text-align: center;
  vertical-align: middle;
  padding: 0px 10px;
  border-width: 0;
  transition: background 0.1s ease-out 0s, box-shadow 0.15s cubic-bezier(0.47, 0.03, 0.49, 1.38) 0s;
  outline: none !important;

  background-color: ${props => getBackgroundColor(props)};
  border-color: ${props => getBackgroundColor(props)};
  color: ${props => getColor(props)};

  &:hover {
    background-color: ${props => getBackgroundHoverColor(props)};
    pointer: cursor;
    text-decoration: ${props => props.buttonStyle === 'link' ? 'underline' : ''};
  }
`;

const Button = (props) => {

  return (
    <StyledButton {...props} />
  )
}

Button.propTypes = {
  buttonStyle: PropTypes.oneOf([
    BUTTON_TYPES.DEFAULT,
    BUTTON_TYPES.PRIMARY,
    BUTTON_TYPES.WARNING,
    BUTTON_TYPES.ERROR,
    BUTTON_TYPES.LINK,
  ]),
  children: PropTypes.node,
};

Button.defaultProps = {
  buttonStyle: BUTTON_TYPES.DEFAULT,
  children: null,
};

export default Button;
