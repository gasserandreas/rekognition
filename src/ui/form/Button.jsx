import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import * as Grommet from 'grommet';

import LoadingIndicator from '../async/LoadingIndicator';

import { Colors } from '../../styles';

export const BUTTON_TYPES = {
  DEFAULT: 'default',
  PRIMARY: 'primary',
  WARNING: 'warning',
  ERROR: 'error',
  LINK: 'link',
};

// calculate background colors
const getBackgroundColor = props => {
  switch (props.buttonStyle) {
    case BUTTON_TYPES.PRIMARY:
      return Colors.Blue.Default;
    case BUTTON_TYPES.WARNING:
      return Colors.Orange.Default;
    case BUTTON_TYPES.ERROR:
      return Colors.Red.Default;
    case BUTTON_TYPES.LINK:
      return 'inherit';
    default:
      return 'rgba(9, 30, 66, 0.04)'
  }
}

const getBackgroundHoverColor = props => {
  switch (props.buttonStyle) {
    case BUTTON_TYPES.PRIMARY:
      return Colors.Blue.Light;
    case BUTTON_TYPES.WARNING:
      return Colors.Orange.Light;
    case BUTTON_TYPES.ERROR:
      return Colors.Red.Light;
    case BUTTON_TYPES.LINK:
      return 'inherit';
    default:
      return 'rgba(9, 30, 66, 0.08)';
  }
}

// text color
const getColor = props => {
  switch (props.buttonStyle) {
    case BUTTON_TYPES.PRIMARY:
    case BUTTON_TYPES.ERROR:
      return Colors.ColorsPalette.White;
    case BUTTON_TYPES.LINK:
      return Colors.Blue.Default;
    default:
      return '#494949';
  }
}

export const StyledLoading = styled(LoadingIndicator)`
  width: 1.2rem;
  margin-left: 0.5rem;
`;

/* istanbul ignore next */
export const StyledButton = styled(Grommet.Button)`
  display: flex;
  align-items: center;
  border-radius: 3px;
  font-weight: 400;
  height: 34px;
  text-align: center;
  vertical-align: middle;
  padding: 0px 10px;
  border-width: 0;
  transition: background 0.1s ease-out 0s, box-shadow 0.15s cubic-bezier(0.47, 0.03, 0.49, 1.38) 0s;
  outline: none !important;

  background-color: ${props => getBackgroundColor(props)};
  border-color: ${props => getBackgroundColor(props)};
  color: ${props => getColor(props)};

  &:hover {
    background-color: ${props => getBackgroundHoverColor(props)};
    pointer: cursor;
    /* istanbul ignore next */
    text-decoration: ${props => props.buttonStyle === 'link' ? 'underline' : ''};
  }
`;

const Button = ({ loading, children, ...props }) => (
  <StyledButton {...props}>
    {children}
    {loading && (
      <StyledLoading
        color={{
          light: Colors.ColorsPalette.White,
          default: Colors.ColorsPalette.White,
          dark: Colors.ColorsPalette.White,
        }}
        circleProps={{
          strokeWidth: 4,
        }}
      />
    )}
  </StyledButton>
);

Button.propTypes = {
  buttonStyle: PropTypes.oneOf([
    BUTTON_TYPES.DEFAULT,
    BUTTON_TYPES.PRIMARY,
    BUTTON_TYPES.WARNING,
    BUTTON_TYPES.ERROR,
    BUTTON_TYPES.LINK,
  ]),
  children: PropTypes.node,
  loading: PropTypes.bool,
};

Button.defaultProps = {
  buttonStyle: BUTTON_TYPES.DEFAULT,
  children: null,
  loading: false,
};

export const __testables__ = {
  getBackgroundColor,
  getBackgroundHoverColor,
  getColor,
}

export default Button;
