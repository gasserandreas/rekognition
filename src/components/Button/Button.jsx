import React from 'react';
import PropTypes from 'prop-types';

import './Button.css';

const Button = (props) => {
  const newProps = {
    ...props,
    className: `button ${props.color} ${props.className ? props.className : ''}`,
  };

  // remove children
  delete newProps.children;


  return (
    <button {...newProps}>{props.children}</button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  color: PropTypes.oneOf(['default', 'blue', 'green']),
};

Button.defaultProps = {
  color: 'default',
};

export default Button;
