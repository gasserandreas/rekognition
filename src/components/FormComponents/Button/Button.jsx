import React from 'react';
import PropTypes from 'prop-types';

import './Button.css';

const Button = (props) => {
  const { className, color, filled } = props;

  const cssClasses = [
    'button',
    className || undefined,
    color,
    filled ? 'filled' : undefined,
  ].filter(item => item !== undefined);

  const newProps = {
    type: 'button',
    tabIndex: 0,
    ...props,
    className: cssClasses.join(' '),
  };

  return (
    <button {...newProps}/>
  );
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  color: PropTypes.oneOf(['warning', 'default']),
  filled: PropTypes.bool,
};

Button.defaultProps = {
  className: undefined,
  color: 'default',
  filled: false,
};

export default Button;
