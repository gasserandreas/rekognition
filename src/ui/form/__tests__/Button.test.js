/* global React, shallow, mount, render, renderer, testSnapshot */

import Button, { StyledButton, StyledLoading } from '../Button';

it('Button should render correctly', () => {
  testSnapshot(<Button>Click me</Button>);
});

it('Button styles should be consistent', () => {
  testSnapshot(<StyledButton />);
})

it('Button should render primary styles', () => {
  testSnapshot(<Button buttonStyle="primary">Click me</Button>);
});

it('Button should render warning styles', () => {
  testSnapshot(<Button buttonStyle="warning">Click me</Button>);
});

it('Button should render error styles', () => {
  testSnapshot(<Button buttonStyle="error">Click me</Button>);
});

it('Button should render link styles', () => {
  testSnapshot(<Button buttonStyle="link">Click me</Button>);
});

// Button loading styles
it('StyledLoading should render correctly', () => {
  testSnapshot(<StyledLoading />);
});

// basic tests
it('Button should render loading component if loading is true', () => {
  const output = mount(<Button />);

  // render without loading
  expect(output.props().loading).toEqual(false);
  expect(output.find(StyledLoading).length).toEqual(0);

  // enable loading
  output.setProps({ loading: true });
  expect(output.props().loading).toEqual(true);
  expect(output.find(StyledLoading).length).toEqual(1);

  // disable loading again
  output.setProps({ loading: false });
  expect(output.props().loading).toEqual(false);
  expect(output.find(StyledLoading).length).toEqual(0);
});
