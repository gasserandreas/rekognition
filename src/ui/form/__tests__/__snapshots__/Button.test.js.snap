// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="sc-htpNat cOCZln StyledButton-sc-323bzc-0 fRAxrl"
  onBlur={[Function]}
  onFocus={[Function]}
  type="button"
>
  Click me
</button>
`;

exports[`Button should render error styles 1`] = `
<button
  className="sc-htpNat jQQDxG StyledButton-sc-323bzc-0 fRAxrl"
  onBlur={[Function]}
  onFocus={[Function]}
  type="button"
>
  Click me
</button>
`;

exports[`Button should render link styles 1`] = `
<button
  className="sc-htpNat eKBEcI StyledButton-sc-323bzc-0 fRAxrl"
  onBlur={[Function]}
  onFocus={[Function]}
  type="button"
>
  Click me
</button>
`;

exports[`Button should render primary styles 1`] = `
<button
  className="sc-htpNat kZVqgB StyledButton-sc-323bzc-0 fRAxrl"
  onBlur={[Function]}
  onFocus={[Function]}
  type="button"
>
  Click me
</button>
`;

exports[`Button should render warning styles 1`] = `
<button
  className="sc-htpNat dZfOOy StyledButton-sc-323bzc-0 fRAxrl"
  onBlur={[Function]}
  onFocus={[Function]}
  type="button"
>
  Click me
</button>
`;

exports[`Button styles should be consistent 1`] = `
<button
  className="sc-htpNat cOCZln StyledButton-sc-323bzc-0 lhIxjU"
  onBlur={[Function]}
  onFocus={[Function]}
  type="button"
/>
`;

exports[`StyledLoading should render correctly 1`] = `
<div
  className="sc-bwzfXH hyhvHp sc-bdVaJa jixwoQ"
  color={
    Object {
      "dark": "#172B4D",
      "default": "#0052CC",
      "light": "#0065FF",
    }
  }
  size={100}
>
  <svg
    className="circular"
    viewBox="25 25 50 50"
  >
    <circle
      className="path"
      cx="50"
      cy="50"
      fill="none"
      r="20"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
  </svg>
</div>
`;
